core/src/main/kotlin/com/tschuchort/compiletesting/KotlinCompilation.kt

```diff
@@ -520,7 +520,7 @@ class KotlinCompilation : AbstractKotlinCompilation<K2JVMCompilerArguments>() {
 
 			val javacCommand = jdkBinFile.absolutePath + File.separatorChar + "javac"
 
-			val isJavac9OrLater = isJavac9OrLater(getJavacVersionString(javacCommand))
+			val isJavac9OrLater = isJavac9OrLater()
```

Comment:
I don't think that this is correct. The system property would always contain the version name of the system JDK or at least the JDK that the program is run with, right? If the `KotlinCompilation` is given a `jdkHome` that points somewhere else, then that JDK's version would be different from the `java.specification.version`.

Reply:
How would one use a different javac than the one in the JDK you're using?

Reply:
Two different JDKs can be in use at the same time: the `processJdk`, that is the JDK of the host process which calls `KotlinCompilation` (where the JUnit tests are) and `jdkHome`, the JDK that is used to actually compile the source files given to the compilation test.  The idea is to be able to use a different JDK for each unit test. For example, imagine you are writing a library that generates classes from a JSON file. For your library project you set Java version 13 to be able to use new language features in the code for your JSON processor, but you want the _generated_ classes to be Java 8 compatible. So your JUnit tests will run under JDK 13 but the `KotlinCompilation` should invoke the javac of JDK 8. If you look at the code here, the `compileJava` method will use the `ToolProvider.getSystemJavaCompiler` (which should also run in the same process if I'm not mistaken) if no specific `jdkHome` is given, but forks a new process `$jdkHome/bin/javac` if a `jdkHome` is specified that is different from the JDK of the host process.

Reply:
Target language version is controlled by the `release` flag, not necessarily the JDK. Kotlin's equivalent is `jvmTarget`. Any JDK can target its current release or older, similar to how kotlinc can target different `jvmTarget` values or even different versions of Kotlin. If you want to customize the JDK home used, kotlinc accepts that as a compiler argument that consumers should use.

The implementation here matches how Kotlin's own compiler internals determine it, at the very least I think it's good to match that for consistency.